api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -156,7 +160,29 @@ private void processOverloadedRuns(final List<PipelineRun> runs) {
                 .filter(pod -> isPodUnderPressure(pod.getValue(), thresholds))
                 .collect(Collectors.toList());
 
+        final List<PipelineRun> runsToUpdateTags = getRunsToUpdatePressuredTags(running, runsToNotify);
         notificationManager.notifyHighResourceConsumingRuns(runsToNotify, NotificationType.HIGH_CONSUMED_RESOURCES);
+        pipelineRunManager.updateRunsTags(runsToUpdateTags);
+    }
+
+    private List<PipelineRun> getRunsToUpdatePressuredTags(final Map<String, PipelineRun> running,
+                              final List<Pair<PipelineRun, Map<ELKUsageMetric, Double>>> runsToNotify) {
+        final Set<Long> runsIdToNotify = runsToNotify
+                .stream()
+                .map(p -> p.getLeft().getId())
+                .collect(Collectors.toSet());
+        final Predicate<PipelineRun> isToBeNotified = r -> runsIdToNotify.contains(r.getId());
```

Comment:
also do we need this? I believe that `runsIdToNotify.contains` also understandable